services/worker-manager/test/provisioner_test.js

```diff
@@ -358,6 +358,39 @@ helper.secrets.mockSuite(testing.suiteName(), [], function(mock, skipping) {
           Severity: LEVELS.info,
         });
     });
+
+    test("provision loop is not running in parallel", async function() {
+      await WorkerPool.fromApi({
+        workerPoolId: 'pp/ww',
+        providerId: 'testing1',
+        previousProviderIds: ['NO-SUCH'],
+        description: '',
+        created: new Date(),
+        lastModified: new Date(),
+        config: {},
+        owner: 'me@example.com',
+        emailOnError: false,
+        providerData: {},
+      }).create(helper.db);
+      const provisioner = await helper.load('provisioner');
+      try {
+        await Promise.all([
+          provisioner.provision(),
+          provisioner.provision(),
+        ]);
+      } catch (err) {
+        if (err.message !== 'provision loop interference') {
+          throw err;
```

Comment:
I've always used something like `assert.fail(err)` instead of throwing the `err`.

As long as throwing works the same and fails the test properly, I'm fine with what you have here.

Reply:
ah yeah, it should be something like `assert.throws()` can be changed. Also used same test suite for guidance :)

Reply:
`assert.throws()` is asserting that some error will be thrown in the call. I don't think that's what you want here since you're just wanting the test itself to fail if the error isn't `provision loop interference`, so I think you need `assert.fail(err)` here.

Reply:
I think I've figured it finally out. I am testing if the error did happen. So here `assert.rejects` is working exactly as it should. Please check updated code